pubsub/pubsub_goredis.go

```diff
@@ -124,30 +125,27 @@ func (ps *GoRedisPubSub) Subscribe(ctx context.Context, topic string) Subscripti
 		for {
 			select {
 			case <-sub.done:
-				close(sub.ch)
+				sub.mut.Lock()
+				sub.cb = nil
```

Comment:
Why do we want to make sure the `cb` is cleared when `sub.Close` is called?
If we don't clear `sub.cb` here, we could get rid of the mutex all together

Reply:
The point of the sub.cb stuff is to make it possible to close a subscription and not receive any more messages. If we don't care to ever let someone do that, all of this could get simpler.

Reply:
I see. Once Close is called, the goroutine started in Subscribe will exit when the sub.done case is hit in the select statement. That will also set the sub.cb to nil. I'm trying to understand how sub.cb would get called again after the goroutine has existed